src/modal/index.tsx

```diff
@@ -58,43 +108,44 @@ export const Modal: React.FC<ModalProps> = ({
 }) => {
   const { doCloseModal } = useModalState();
   const ref = React.useRef(null);
+
+  React.useEffect(() => {
+    const func = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        if (close) {
+          close();
+        } else {
+          doCloseModal();
+        }
+      }
+    };
+
+    if (isOpen) {
+      typeof window !== 'undefined' &&
+        window.document &&
+        window.document.createElement &&
```

Comment:
Why the need for checking `createElement`?

Reply:
pulled it from https://github.com/JedWatson/exenv/blob/master/index.js#L14, might not actually need it. Happy to remove.

Reply:
Let's keep it, so that some dev one day will do the same thing, and when they ask "Why this?" they can say they got it from us. And the rabbit hole continues...